src/python_testing/TC_CADMIN_1_5.py

```diff
@@ -44,13 +45,59 @@
 
 
 class TC_CADMIN_1_5(MatterBaseTest):
-    async def get_txt_record(self):
+    async def get_all_txt_records(self):
         discovery = mdns_discovery.MdnsDiscovery(verbose_logging=True)
-        comm_service = await discovery.get_commissionable_service(
-            discovery_timeout_sec=240,
-            log_output=False,
-        )
-        return comm_service
+        discovery._service_types = [mdns_discovery.MdnsServiceType.COMMISSIONABLE.value]
+        await discovery._discover(discovery_timeout_sec=240, log_output=False)
+
+        if mdns_discovery.MdnsServiceType.COMMISSIONABLE.value in discovery._discovered_services:
+            return discovery._discovered_services[mdns_discovery.MdnsServiceType.COMMISSIONABLE.value]
+        return []
+
+    async def wait_for_correct_cm_value(self, expected_cm_value, expected_discriminator, max_attempts=5, delay_sec=5):
+        """Wait for the correct CM value and discriminator in DNS-SD with retries."""
+        for attempt in range(max_attempts):
+            services = await self.get_all_txt_records()
+
+            # Look through all services for a match
+            for service in services:
+                cm_value = service.txt_record.get('CM')
+                d_value = service.txt_record.get('D')
+
+                # Convert to strings for comparison
+                if str(cm_value) == str(expected_cm_value) and str(d_value) == str(expected_discriminator):
```

Comment:
Why do we convert to strings? Would it make sense to type safe one of them and only correct the other?

Should cm and discriminator be integers and we should try to convert to ints instead?

Reply:
```suggestion
                if cm_value == str(expected_cm_value) and d_value == str(expected_discriminator):
```

Is there any reason why cm_value or d_value would not be strings? You could also use a `get(..., 'MISSING')` as you do below.

Reply:
Using the dataclass method was definitely the way to go, the above code is no longer needed and has been removed.